### PR TITLE
Support multiple template languages

### PR DESCRIPTION
**Pull request by dNitza**
This PR adds support for multiple template engines to be used per format in each view.

e.g.

``` ruby
my_view.call(format: [:erb, :slim])
```

This shouldn't be a breaking change since `options[:engines]` is wrapped in an array so:

``` ruby
my_view.call(format: :slim)
```

is still valid.

There is currently  a limitation on trying to render a template with the same name but with a different engine.

Given:

``` ruby
my_view.call(format: [:erb, :slim])
```

and

```
.
├── users.html.slim
└── users.html.erb
```

The `.erb` template would take precedence over the `.slim` template as it was the first specified under `format`.

### CODE REVIEW

File: lib/dry/view/renderer.rb [hunk 1]
```diff
@@ -4,11 +4,11 @@
 module Dry
   module View
     class Renderer
-      include Dry::Equalizer(:paths, :format, :engine)
+      include Dry::Equalizer(:dir, :root, :engines)
 
       TemplateNotFoundError = Class.new(StandardError)
 
-      attr_reader :paths, :format, :engine, :tilts
```
Comment (timriley) [inline lib/dry/view/renderer.rb]: Oops, I think we want to keep `:paths` both here and above. We've switched from `:root` to `:paths` so we can support multiple view template paths.

File: lib/dry/view/renderer.rb [hunk 2]
```diff
@@ -17,7 +17,7 @@ def self.tilts
       def initialize(paths, options = {})
         @paths = paths
         @format = options.fetch(:format)
-        @engine = options.fetch(:engine)
```
Comment (timriley) [inline lib/dry/view/renderer.rb]: We'd want to keep the `.fetch` behaviour here?

File: lib/dry/view/renderer.rb [hunk 3]
```diff
@@ -39,21 +39,23 @@ def render(path, scope, &block)
...
+        engines.map do |engine|
+          template_name = "#{name}.#{format}.#{engine}"
 
-        paths.inject(false) { |result, path|
-          result || path.lookup(template_name)
-        }
+          paths.inject(false) { |result, path|
+            result || path.lookup(template_name)
+          }
+        end.reject{|f| f == false}.first
```
Comment (timriley) [inline lib/dry/view/renderer.rb]: Could we make it so this outer loop "short-circuit" subsequent lookups if we've already found a template? This is what I was doing with the inject and `result || lookup` stuff in that inner loop...

File: lib/dry/view/renderer.rb [hunk 4]
```diff
@@ -39,21 +39,23 @@ def render(path, scope, &block)
...
       end
 
       private
 
       def tilt(path)
-        tilts.fetch(path) { tilts[path] = Tilt[engine].new(path, nil, default_encoding: "utf-8") }
+        tilts.fetch(path) { tilts[path] = Tilt.new(path, nil, default_encoding: "utf-8") }
```
Comment (timriley) [inline lib/dry/view/renderer.rb]: Interesting... we don't actually need to specify an engine for Tilt here?

Comment (dNitza) [inline lib/dry/view/renderer.rb]: I've let `Tilt` infer the type of template based on the extension of the file we give it. We can do something like `Tilt[path.split('.').pop]` to be explicit.

AFAICT there isn't a benefit to either approach?: https://github.com/rtomayko/tilt#basic-usage

Comment (timriley) [inline lib/dry/view/renderer.rb]: Letting it infer the template type seems fine to me here. Means we can continue to create only a single tilt instance.

**Comment by dNitza** [comment]
Have update this PR with your feedback notes @timriley :)

File: lib/dry/view/renderer.rb (hunk 4 above)
Comment (dNitza) [inline lib/dry/view/renderer.rb]: Letting Tilt infer the template engine could cause issues in a threaded environment

> It's recommended that calling programs explicitly require template engine libraries (like 'erb' above) at load time. Tilt attempts to lazy require the template engine library the first time a template is created but this is prone to error in threaded environments.

**Comment by timriley** [comment]
Thanks for forging the path with this, @dNitza! 🙏 I'll close this now in favour of #27.

**Pull request closed without merging**